File: app/src/main/java/com/divercity/app/core/base/SharedPreferencesManager.java
package com.divercity.app.core.base;

import android.content.Context;
import android.content.SharedPreferences;

public class SharedPreferencesManager<T extends Enum> {

    private SharedPreferences mPref;

    public SharedPreferencesManager(Context context, String prefName) {
        mPref = context.getSharedPreferences(prefName, Context.MODE_PRIVATE);
    }

    private SharedPreferences.Editor getEditor() {
        return mPref.edit();
    }

    public void clearAllData() {
        getEditor().clear().commit();
    }

    //    PUTS

    public void put(T key, String val) {
        getEditor().putString(key.name(), val).commit();
    }

    public void put(T key, int val) {
        getEditor().putInt(key.name(), val).commit();
    }

    public void put(T key, long val) {
        getEditor().putLong(key.name(), val).commit();
    }

    //    GETS

    public int getInt(T key) {
        return mPref.getInt(key.name(), -1);
    }

    public String getString(T key, String defaultValue) {
        return mPref.getString(key.name(), defaultValue);
    }

    public String getString(T key) {
        return mPref.getString(key.name(), null);
    }

    public long getLong(T key) {
        return mPref.getLong(key.name(), -1);
    }

}

File: app/src/main/java/com/divercity/app/core/sharedpref/SharedPreferencesManager.kt
package com.divercity.app.core.sharedpref

import android.content.Context
import android.content.SharedPreferences

/**
 * Created by lucas on 24/10/2018.
 */

class SharedPreferencesManager<T : Enum<*>>(context: Context, prefName: String) {

    private val mPref: SharedPreferences = context.getSharedPreferences(prefName, Context.MODE_PRIVATE)

    private val editor: SharedPreferences.Editor
        get() = mPref.edit()

    fun clearAllData() = editor.clear().commit()

    //    PUTS

    fun put(key: T, value: String) = editor.putString(key.name, value).commit()

    fun put(key: T, value: Int) = editor.putInt(key.name, value).commit()

    fun put(key: T, value: Long) = editor.putLong(key.name, value).commit()

    //    GETS

    fun getInt(key: T): Int = mPref.getInt(key.name, -1)

    fun getString(key: T, defaultValue: String): String? = mPref.getString(key.name, defaultValue)

    fun getString(key: T): String? = mPref.getString(key.name, null)

    fun getLong(key: T): Long = mPref.getLong(key.name, -1)
}